Add language changes to beta1 release note (#416)
ballerina-platform/ballerina-release: release-notes/swan-lake-beta1-release-note.md

### Language Updates

#### Improvements

- Worker deadlock detection has been improved to include the `wait` action.

> - `self` of an isolated object no longer needs to be accessed within a `lock` statement unless it is used to access a field that is either not `final` or is not a subtype of `readonly` or `isolated object`.

Reviewer: ```suggestion
- The`self` of an isolated object no longer needs to be accessed within a `lock` statement unless it is used to access a field that is either not `final` or is not a subtype of a `readonly` or `isolated object`.
```

Reviewer 2: We have to say "The `self` variable" if we are using "The".

Why do we need the "a" before "`readonly` or `isolated object`". It's two different types here, like saying "not a subtype of `T1` or `T2`."

```suggestion
- The `self` variable of an isolated object no longer needs to be accessed within a `lock` statement unless it is used to access a field that is either not `final` or is of a type that is not a subtype of `readonly` or `isolated object`.
```

> - `xml:strip()` does not mutate the XML value, thus using `xml:strip()` on immutable XML values is allowed.

Reviewer: ```suggestion
- The `xml:strip()` does not mutate the XML value, and thereby, using the `xml:strip()` on immutable XML values is allowed.
```

Reviewer 2: Same comment regarding "the". The content within backticks in these contexts usually refer to names, so we can't just say "The `x` does not...". I think we should use therefore instead of thereby here.

```suggestion
- The `lang.xml:strip()` function does not mutate an XML value. Therefore, using `lang.xml:strip()` on immutable XML values is allowed.
```

Reviewer: +1

#### Breaking Changes

- An included record parameter of a function can only be specified after any required and/or defaultable parameters of the function.
- Additive expressions and multiplicative expressions are no longer supported with numeric values when the static types of the operands belong to different numeric basic types.

> - Configurable variables are implicitly `final` now. Moreover, the type of such a variable is now effectively the intersection of the specified type and `readonly`. Therefore configurable variables no longer support the `final` and `isolated` qualifiers.

Reviewer: ```suggestion
- Configurable variables are implicitly `final` now. Moreover, the type of such a variable is now effectively the intersection of the specified type and `readonly`. Therefore, configurable variables no longer support the `final` and `isolated` qualifiers.
```

> - Type narrowing will no longer take place for captured variables of an anonymous function, since the narrowed type cannot be guaranteed during the execution of the function.

Reviewer: ```suggestion
- Type narrowing will no longer take place for captured variables of an anonymous function since the narrowed type cannot be guaranteed during the execution of the function.
```

- The `lang.xml:strip()` function does not mutate an XML value. Therefore, using the `lang.xml:strip()` function on immutable XML values is allowed.

#### Breaking Changes

- An included record parameter of a function can only be specified after any required and/or defaultable parameters of the function.
- Additive expressions and multiplicative expressions are no longer supported with numeric values when the static types of the operands belong to different numeric basic types.
- Configurable variables are implicitly `final` now. Moreover, the type of such a variable is now effectively the intersection of the specified type and `readonly`. Therefore, configurable variables no longer support the `final` and `isolated` qualifiers.
- Type narrowing will no longer take place for captured variables of an anonymous function since the narrowed type cannot be guaranteed during the execution of the function.
- Type narrowing will now be reset after a compound assignment.
- Worker message passing after waiting for the same worker has been disallowed.
- When a named worker is used in a `wait` action, it can no longer be used in a variable reference anywhere else.

> - When the `type-descriptor` is ambiguous, it is considered according to the following table, in which the type precedence is presented in the decreasing order.

Reviewer 2: type-descriptor is not a keyword.

```suggestion
- When the type-descriptor is ambiguous, it is parsed according to the following table, in which the precedence is specified in decreasing order.
```

@lochana-chathura it's OK to say parsed here, right?

Reviewer 3: +1

#### Improvements

#### Breaking Changes

> - `io.ballerina.runtime.api.types.Type#getName` and `io.ballerina.runtime.api.types.Type#getQualifiedName` now returns an empty string if no name was associated with the type. The `io.ballerina.runtime.api.types.Type#toString` method can be used to get the string representation of a type if required.

Reviewer: ```suggestion
- The `io.ballerina.runtime.api.types.Type#getName` and `io.ballerina.runtime.api.types.Type#getQualifiedName` now return an empty string if no name was associated with the type. The `io.ballerina.runtime.api.types.Type#toString` method can be used to get the string representation of a type if required.
```

Reviewer 2: ```suggestion
- The `io.ballerina.runtime.api.types.Type#getName` and `io.ballerina.runtime.api.types.Type#getQualifiedName` methods now return an empty string if no name was associated with the type. The `io.ballerina.runtime.api.types.Type#toString` method can be used to get the string representation of a type if required.
```

> - Wait actions that wait for expressions that are not named workers can return errors now. The eventual type of such wait future expressions is now `T|error`, `T` being the type of the original return value.

Reviewer: ```suggestion
- The` wait` actions that wait for expressions that are not named workers can return errors now. The eventual type of such wait future expressions is now `T|error`in which `T` being the type of the original return value.
```

Reviewer 2: Any particular reason to change from "Wait actions"? Do we really need the "the"?

Also isn't "where" more suitable that "in which". The spec also uses where.

```suggestion
- Wait actions that wait for expressions that are not named workers can return errors now. The eventual type of such wait future expressions is now `T|error`, where `T` is the type of the original return value.
```

Reviewer: +1

username = "Jane"
location = "US"
```

> #### Breaking Changes

Reviewer 4: ```suggestion
#### Other Changes

- Taint analyzer is disabled and `--taint-check` compiler flag is removed. The annotations `@tainted` and `@untainted` were left intact as place holders for backword compatibility.

#### Breaking Changes
```

username = "Jane"
location = "US"
```
#### Other Changes

> - Taint analyzer is disabled and `--taint-check` compiler flag is removed. The annotations `@tainted` and `@untainted` were left intact as place holders for backword compatibility.

Reviewer 2: Can't we include this also under breaking changes?

```suggestion
- Taint analyzer is disabled and the `--taint-check` option is removed. The annotations `@tainted` and `@untainted` were left intact as place holders for backward compatibility.
```

Reviewer 4: Fixed https://github.com/ballerina-platform/ballerina-release/pull/416/commits/4699828ad153bcf58a0e588b0524654fbe0a5db4

#### Ballerina Packages Updates

### Breaking Changes

> - Taint analyzer is disabled and the `--taint-check` option is removed. The annotations `@tainted` and `@untainted` were left intact as place holders for backward compatibility. Any script that used to pass `--taint-check` option to `bal run/build` commands will fail.

Reviewer 2: ```suggestion
- Taint analyzer is disabled and the `--taint-check` option is removed. The annotations `@tainted` and `@untainted` were left intact as place holders for backward compatibility. Any script that used to pass the `--taint-check` option to the `bal run` or `bal build` commands will fail.
```
